apps/dotcms-ui/src/app/portlets/shared/dot-content-types-listing/dot-content-type.store.ts

```diff
@@ -0,0 +1,38 @@
+import { ComponentStore } from '@ngrx/component-store';
+import { Injectable } from '@angular/core';
+import { Observable } from 'rxjs';
+import { DotCMSAssetDialogData } from '@dotcms/dotcms-models';
+
+export interface ContentTypeState {
+    isVisibleCloneDialog: boolean;
+    assetSelected: DotCMSAssetDialogData | null;
+}
+
+const initialState: ContentTypeState = {
+    isVisibleCloneDialog: false,
+    assetSelected: null
+};
+
+@Injectable()
+export class DotContentTypeStore extends ComponentStore<ContentTypeState> {
+    readonly assetSelected$: Observable<DotCMSAssetDialogData> = this.select(
+        ({ assetSelected }) => assetSelected
+    );
+    readonly isVisibleCloneDialog$: Observable<boolean> = this.select(
+        ({ isVisibleCloneDialog }) => isVisibleCloneDialog
+    );
+    readonly showCloneDialog = this.updater((state, assetSelected: DotCMSAssetDialogData) => ({
+        ...state,
+        assetSelected,
+        isVisibleCloneDialog: true
+    }));
+    readonly hideCloneDialog = this.updater((state) => ({
+        ...state,
+        assetSelected: null,
+        isVisibleCloneDialog: false
```

Comment:
Do we need this flag or can you use `assetSelected` as flag?

Reply:
I put isVisibleCloneDialog thinking in the future if we need add more dialogs. It trigger the Clone Dialog